app/javascript/dashboard/components/ui/DatePicker/DatePicker.vue

```diff
@@ -0,0 +1,280 @@
+<script setup>
+import { ref, watch } from 'vue';
+import {
+  getActiveDateRange,
+  moveCalendarDate,
+} from './helpers/DatePickerHelper';
+import {
+  format,
+  isValid,
+  startOfMonth,
+  subDays,
+  startOfDay,
+  endOfDay,
+  isBefore,
+  subMonths,
+  addMonths,
+  isSameMonth,
+  differenceInCalendarMonths,
+  setMonth,
+  setYear,
+  isAfter,
+} from 'date-fns';
+
+import DatePickerButton from './components/DatePickerButton.vue';
+import CalendarDateInput from './components/CalendarDateInput.vue';
+import CalendarDateRange from './components/CalendarDateRange.vue';
+import CalendarYear from './components/CalendarYear.vue';
+import CalendarMonth from './components/CalendarMonth.vue';
+import CalendarWeek from './components/CalendarWeek.vue';
+import CalendarFooter from './components/CalendarFooter.vue';
+
+const showDatePicker = ref(false);
+const calendarViews = ref({ start: 'week', end: 'week' });
+const currentDate = ref(new Date());
+const startCurrentDate = ref(startOfDay(currentDate.value)); // Today's date at the start of the day (starts the current month)
+const endCurrentDate = ref(addMonths(startOfDay(currentDate.value), 1)); // One month ahead of today at the start of the day (starts the next month)
+const selectedStartDate = ref(startOfDay(subDays(currentDate.value, 6)));
+const selectedEndDate = ref(endOfDay(currentDate.value));
+const selectingEndDate = ref(false);
+const selectedRange = ref('last7days');
+const hoveredEndDate = ref(null);
+
+const manualStartDate = ref('');
+const manualEndDate = ref('');
+
+const emit = defineEmits(['change']);
+
+watch(selectedRange, newRange => {
+  if (newRange !== 'custom') {
+    const isLast7days = newRange === 'last7days';
+    startCurrentDate.value = selectedStartDate.value;
+    endCurrentDate.value =
+      isLast7days && isSameMonth(selectedStartDate.value, selectedEndDate.value)
+        ? startOfMonth(addMonths(selectedStartDate.value, 1))
+        : selectedEndDate.value;
+    selectingEndDate.value = false;
+  } else if (!selectingEndDate.value) {
+    startCurrentDate.value = startOfDay(currentDate.value);
+  }
+});
```

Comment:
Let's document this heavily, and other watch functions too. Right now you have all the context, but in the future when debugging edge-cases with this, or making any changes, we might not have that.